core/src/main/java/nu/yona/server/subscriptions/entities/NewDeviceRequest.java

```diff
@@ -0,0 +1,83 @@
+package nu.yona.server.subscriptions.entities;
+
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.Transient;
+
+import nu.yona.server.crypto.CryptoSession;
+import nu.yona.server.crypto.StringFieldEncrypter;
+import nu.yona.server.entities.EntityWithID;
+import nu.yona.server.entities.RepositoryProvider;
+
+/*
+ * A request to add another device for an existing user. The data cannot be encrypted with the user 'password' or auto-generated
+ * key because that is stored on the device and cannot be entered by the user. Therefore we have to transfer the user 'password'
+ * to the new device and obviously encrypt it (with a secret entered by the user, e.g. her PIN or a one time password) while
+ * transferring.
+ */
+@Entity
+@Table(name = "NEW_DEVICE_REQUESTS")
+public class NewDeviceRequest extends EntityWithID
+{
+	public static NewDeviceRequestRepository getRepository()
+	{
+		return (NewDeviceRequestRepository) RepositoryProvider.getRepository(NewDeviceRequest.class, UUID.class);
+	}
+
+	private Date expirationDateTime;
```

Comment:
The app doesn't need to know when the request expires. We won't show that
to the user. The cleanup process should use the configuration setting to
see what to remove. Storing the expiration date is like storing someone's
age instead of birthday.

Groetjes, Bert
On Nov 19, 2015 2:15 PM, "aristotelos" <notifications@github.com> wrote:

> In
> core/src/main/java/nu/yona/server/subscriptions/entities/NewDeviceRequest.java
> <https://github.com/yonadev/yona-server/pull/14#discussion_r45338349>:
>
> > +/*
> > + * A request to add another device for an existing user. The data cannot be encrypted with the user 'password' or auto-generated
> > + * key because that is stored on the device and cannot be entered by the user. Therefore we have to transfer the user 'password'
> > + * to the new device and obviously encrypt it (with a secret entered by the user, e.g. her PIN or a one time password) while
> > + * transferring.
> > + */
> > +@Entity
> > +@Table(name = "NEW_DEVICE_REQUESTS")
> > +public class NewDeviceRequest extends EntityWithID
> > +{
> > +	public static NewDeviceRequestRepository getRepository()
> > +	{
> > +		return (NewDeviceRequestRepository) RepositoryProvider.getRepository(NewDeviceRequest.class, UUID.class);
> > +	}
> > +
> > +	private Date expirationDateTime;
>
> But then you get a difference between the DTO and the entity, which is not
> nice. Moreover the cleanup policy needs the expiration date to cleanup.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yonadev/yona-server/pull/14/files#r45338349>.
>